### DISCUSSION TITLE
Connection error Raspberry PI 4

### DISCUSSION THREAD

**Discussion by stefano-gentile** [Chia-Network/chia-blockchain#5886] [Support]
Hi guys, I cant sort out this connection problem. My setup is a Raspberry pi 4 8GB RAM as my farming/harvester (UBUNTU Server OS) and my main PC (WINDOWS 10) which is full node, plotting and farming.

I've already done this [Farming on many machines · Chia-Network/chia-blockchain Wiki · GitHub] https://github.com/Chia-Network/chia-blockchain/wiki/Farming-on-many-machines

When run `chia start harvester` on the Raspberry it seems like everything is ok, in main PC shows Last attempted proof 0/54 (my plots connected on the Raspberry), but if run `chia farm summary` on the RasPI, I get this

![asasas](https://user-images.githubusercontent.com/26451058/118155970-2f79d000-b3ef-11eb-82f0-b4ab37b60b7e.png)

Ports are open on the main PC and upnp is false on RasPI, I don't know if am I forgetting sth.

Btw, thanks

**Comment by sargonas** [comment]
So real quick to clarify based on the above:

You have a farmer running on *both* the windows 10 pc with the full node AND on the pi... with a harvester on each of those devices as well? Are each of those harvesters talking to their respective farmer on the same device, and both devices talking to the windows 10 node? or... at least... is that the *intended* way they should be working?

Let me know and we can try to poke at some stuff from there, but I will lead off with if that IS the case, is there a particular reason you chose to not have the pi harvester use the win10 farmer as it's authoritative farmer? having less farmers in the mix might help reduce room for issues, for a start, if so.

**Comment by Zeesy** [comment]
I'm experiencing similar issues, and believe it's because I haven't figured out how to run the pi off the win10 full node. Chia harvester is running. Here's some results:
```
chia show -a host:port
failed to connect to host:port
```

```
chia farm summary
Connection error. Check if wallet is running at 9256
Connection error. Check if wallet is running at 8559
```

```
chia wallet show
Connection error. Check if wallet is running at 9256
```

I set my pi up using [this tutorial](https://medium.com/geekculture/how-to-use-your-raspberry-pi-as-a-chia-harvester-66440c17d318).

**Comment by stefano-gentile** [comment]
> I'm experiencing similar issues, and believe it's because I haven't figured out how to run the pi off the win10 full node. Chia harvester is running. Here's some results:
> 
> ```
> chia show -a host:port
> failed to connect to host:port
> ```
> 
> ```
> chia farm summary
> Connection error. Check if wallet is running at 9256
> Connection error. Check if wallet is running at 8559
> ```
> 
> ```
> chia wallet show
> Connection error. Check if wallet is running at 9256
> ```
> 
> I set my pi up using [this tutorial](https://medium.com/geekculture/how-to-use-your-raspberry-pi-as-a-chia-harvester-66440c17d318).

I think that I was confused.. If u just want to use the PI as a harvester and check that everything is working, you should open config.yaml and write INFO in log_level

e.g:
 ` logging: &id001
    log_filename: log/debug.log
    log_level: INFO
    log_stdout: false`

Then copy this on the Terminal to check if is working `cat  ~/.chia/mainnet/log/debug.log`

The output should be like this `2021-05-22T21:54:02.898 harvester chia.harvester.harvester: INFO [number] plots were eligible for farming [key]` Time: 0.23380 s. Total x plots`

From what I know `chia show -a` or `chia farm summary ` is used if you have your PI configured as Full Node, Wallet or Farmer.

**Reply by sargonas** [reply]
You've got the right of it. If it's JUST a harvester, then there is two things to bear in mind:
1. you're going to edit the config under the `harvester:` section and point it at the IP of your farmer. (You don't have to worry about node configurations since the Farmer takes care of that on it's side)
2. Running a 'chia farm summary` on a remote harvester itself or on a centralized farmer with remote harvesters, has some quirks at this time where all the data does not report back as well as it does if you run it on a system where the farmer and harvester are all in one. This is a known issue and on the backlog to address at some point.

**Reply by Zeesy** [reply]
It seems that some people are able to view their plots in the main machine GUI, but limited only to a line in Last Attempted Proof. Ie.
![image](https://user-images.githubusercontent.com/17345935/119483951-881c6780-bd23-11eb-8f81-aacef652e8a9.png)
(from[ the tutorial](https://medium.com/geekculture/how-to-use-your-raspberry-pi-as-a-chia-harvester-66440c17d318) I was following).
My `harvester` section is connected to the correct IPv4 address, but I do not see the Pi entry in my main machine GUI and `cat ~/.chia/mainnet/log/debug.log` shows that my harvester is stuck in "Reconnecting". Port 8444 has port forwarding turned on. Is my harvester just not connected to the main machine? If so, where should I check to fix that? If it is connected, what would be the command to check status?

**Reply by Zeesy** [reply]
any updates on this?

**Reply by loppefaaret** [reply]
a harvester connects to a farmer, on port 8447, make sure this port is opened to accept the connection from the harvester

**Comment by BrandtH22** [comment]
@stefano-gentile , since this ticket was opened a number of years ago without further comments it seems this issue has been resolved. 
We will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions or comments.

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
